kiwi/config/functions.sh

```diff
@@ -1243,4 +1243,37 @@ function baseQuoteFile {
     mv "${conf}" "${file}"
 }
 
+#======================================
+# vagrantSetup
+#--------------------------------------
+function vagrantSetup {
+    # This function configures the image to work as a vagrant box.
+    # These are the following steps:
+    # - add the vagrant user to /etc/sudoers
+    # - insert the insecure vagrant ssh key
+    # - create the default /vagrant share
+    # - apply some recommended ssh settings
+
+    # insert the default insecure ssh key from here:
+    # https://github.com/hashicorp/vagrant/blob/master/keys/vagrant.pub
+    mkdir -p /home/vagrant/.ssh/
+    echo "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkTkyrtvp9eWW6A8YVr+kz4TjGYe7gHzIw+niNltGEFHzD8+v1I2YJ6oXevct1YeS0o9HZyN1Q9qgCgzUFtdOKLv6IedplqoPkcmF0aYet2PkEDo3MlTBckFXPITAMzF8dJSIFo9D8HfdOV0IAdx4O7PtixWKn5y2hMNG0zQPyUecp4pzC6kivAIhyfHilFR61RGL+GPXQ2MWZWFYbAGjyiYJnAmCP3NOTd0jMZEnDkbUvxhMmBYSdETk1rRgm+R4LOzFUGaHqHDLKLX+FIPKcF96hrucXzcWyLbIbEgE98OHlnVYCzRdK8jlqm8tehUc9c9WhQ== vagrant insecure public key" > /home/vagrant/.ssh/authorized_keys
+    chmod 0600 /home/vagrant/.ssh/authorized_keys
+    chown -R vagrant:vagrant /home/vagrant/
+
+    # recommended ssh settings for vagrant boxes
+    echo "UseDNS no" >> /etc/ssh/sshd_config
+    echo "GSSAPIAuthentication no" >> /etc/ssh/sshd_config
+
+    # vagrant assumes that it can sudo without a password
+    echo "vagrant ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
+
+    # the default shared folder
+    mkdir -p /vagrant
+    chown -R vagrant:vagrant /vagrant
+
+    /usr/sbin/sshd-gen-keys-start
+    suseInsertService sshd
+}
+
```

Comment:
I can drop it, I thought I'll add it, since this is pretty much the standard way how a vagrant box should look and everyone will essentially copy and paste this into their `config.sh`. Nevertheless, this does not need to be part of kiwi.